compute_tools/src/compute.rs

```diff
@@ -583,32 +587,59 @@ impl ComputeNode {
             pspec.spec.remote_extensions,
         );
 
-        // Launch remaining service threads
-        let _monitor_handle = launch_monitor(self);
-        let _configurator_handle = launch_configurator(self);
+        ////// PRE-STARTUP PHASE: things that need to be finished before we start the Postgres process
+
+        // Collect all the tasks that must finish here
+        let mut pre_tasks = tokio::task::JoinSet::new();
+
+        // If there are any remote extensions in shared_preload_libraries, start downloading them
+        self.download_preload_extensions(pspec, &mut pre_tasks);
+
+        // Prepre pgdata directory. This downloads the basebackup, among other things.
+        let this = self.clone();
+        let cs = compute_state.clone();
+        let sp = start_compute_span.clone();
+        pre_tasks.spawn_blocking(move || {
+            let _e = sp.enter();
+            this.prepare_pgdata(&cs)
+        });
```

Comment:
Can you move this into a block, `{ }` so the variables don't leak?

Reply:
Ah yes, this is a bit ugly. I wish there was a better syntax in rust to avoid typing out these clones. 

Aside from the clones, I had some repetition with passing the tracing span to each spawned thread. I refactored that into a little helper trait; it makes these callers a little less verbose. Please take a look.